pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -623,6 +625,74 @@ func getVMIMigrationDataSize(vmi *v1.VirtualMachineInstance) int64 {
 	return memory.ScaledValue(resource.Giga)
 }
 
+func tuneMigrationCompression(vmi *v1.VirtualMachineInstance, params *libvirt.DomainMigrateParameters) {
+	switch compressMethod := vmi.Annotations[v1.LiveMigrationCompressionMethod]; compressMethod {
+	case "mt":
+		params.Compression = "mt"
+		params.CompressionSet = true
+
+		//set compression level if needed
+		compressLevel := vmi.Annotations[v1.LiveMigrationThreadedCompressionLevel]
+		level, err := strconv.Atoi(compressLevel)
+		if err != nil {
+			log.Log.Reason(err).Warning("failed to set live migration compression level")
+			return
```

Comment:
should we be preventing the migration from occurring if these parameters aren't set right by returning the error ?

For testing, this might result in some tricky scenarios where the migration ignores the parameters you thought it was using. The only way you'd know it happened is if you looked at the logs.